### PR TITLE
Make `CorruptedTaskHistoryIntegrationTest` less flaky

### PR DESCRIPTION
**Pull request by lacasseio**
### Context
Fixes https://github.com/gradle/gradle-private/issues/1594

### Gradle Core Team Checklist
- [ ] Verify design and implementation 
- [ ] Verify test coverage and [CI build status](https://builds.gradle.org/project.html?projectId=Gradle_Check&tab=projectOverview&branch_Gradle_Check=lacasseio%2Ffixit%2F1594)
- [ ] Verify documentation
- [ ] Recognize contributor in release notes

### CODE REVIEW

File: subprojects/core/src/integTest/groovy/org/gradle/api/internal/changedetection/CorruptedTaskHistoryIntegrationTest.groovy [hunk 1]
```diff
@@ -16,34 +16,69 @@
...
         def totalNumberOfOutputFiles = numberOfTasks * numberOfOutputFilesPerTask + totalNumberOfOutputDirectories
 
-        setupTestProject(numberOfFiles, numberOfInputProperties, numberOfTasks, killPollInterval)
+        setupTestProject(numberOfFiles, numberOfInputProperties, numberOfTasks)
 
         executer.beforeExecute {
             // We need a separate JVM in order not to kill the test JVM
             requireGradleDistribution()
+            requireIsolatedDaemons()
+            requireDaemon()
```
Comment (adammurdoch) [inline subprojects/core/src/integTest/groovy/org/gradle/api/internal/changedetection/CorruptedTaskHistoryIntegrationTest.groovy]: I think we can get rid of the `requireGradleDistribution()` above as this shouldn't be true. The fixtures can start a daemon without needing a distribution.

File: subprojects/core/src/integTest/groovy/org/gradle/api/internal/changedetection/CorruptedTaskHistoryIntegrationTest.groovy [hunk 2]
```diff
@@ -87,9 +130,15 @@ class CreateFiles implements Runnable {
 
     @Override
     void run() {
+        if (Boolean.valueOf(System.getProperty("killMe"))) {
```
Comment (adammurdoch) [inline subprojects/core/src/integTest/groovy/org/gradle/api/internal/changedetection/CorruptedTaskHistoryIntegrationTest.groovy]: we don't need the conditional logic, the build is always run this way

File: subprojects/core/src/integTest/groovy/org/gradle/api/internal/changedetection/CorruptedTaskHistoryIntegrationTest.groovy [hunk 3]
```diff
@@ -87,9 +130,15 @@ class CreateFiles implements Runnable {
...
+            ${server.callFromBuildUsingExpression('"worker-for-${outputDir.name}"')}
+        }
         (1..$numberOfFiles).each {
             new File(outputDir, "\${it}.txt").text = "output\${it}"
         }
+        if (Boolean.valueOf(System.getProperty("killMe"))) {
```
Comment (adammurdoch) [inline subprojects/core/src/integTest/groovy/org/gradle/api/internal/changedetection/CorruptedTaskHistoryIntegrationTest.groovy]: Same here

File: subprojects/core/src/integTest/groovy/org/gradle/api/internal/changedetection/CorruptedTaskHistoryIntegrationTest.groovy (hunk 2 above)
Comment (wolfs) [reply subprojects/core/src/integTest/groovy/org/gradle/api/internal/changedetection/CorruptedTaskHistoryIntegrationTest.groovy]: It is not. The first run is executed without any killing.

File: subprojects/core/src/integTest/groovy/org/gradle/api/internal/changedetection/CorruptedTaskHistoryIntegrationTest.groovy [hunk 4]
```diff
@@ -16,34 +16,69 @@
...
+        workerHandler.waitForAllPendingCalls()
+
+        // Release only a few of them
+        workerHandler.release(numberOfExecutedTaskBeforeKilling)
+
+        // Wait until workers finish their jobs
+        killHandler.waitForAllPendingCalls()
+
+        // Kill the daemon
+        killRunningDaemon()
```
Comment (wolfs) [inline subprojects/core/src/integTest/groovy/org/gradle/api/internal/changedetection/CorruptedTaskHistoryIntegrationTest.groovy]: ❌ The idea is to kill the daemon while some tasks are writing their task history, not after they produced the output files, before the task finishes. So in its current state, the test won't do anything anymore.

Comment (lacasseio) [reply subprojects/core/src/integTest/groovy/org/gradle/api/internal/changedetection/CorruptedTaskHistoryIntegrationTest.groovy]: How do we block mid-way writing the task history?

Comment (wolfs) [reply subprojects/core/src/integTest/groovy/org/gradle/api/internal/changedetection/CorruptedTaskHistoryIntegrationTest.groovy]: We can't 😞 
The idea of the test was (and it did reproduce the original problem back then), to kill the daemon while enough tasks were writing the task history, so we would sometimes catch one of those tasks. We can achieve the same with blocking http server I guess.

**Pull request closed without merging**